source/simulator/parameters.cc

```diff
@@ -1784,6 +1789,29 @@ namespace aspect
       AssertThrow (normalized_fields.size() <= n_compositional_fields,
                    ExcMessage("Invalid input parameter file: Too many entries in List of normalized fields"));
 
+      std::vector<std::string> x_compositional_field_types
+        = Utilities::split_string_list
+          (prm.get ("Types of fields"));
+
+      AssertThrow ((x_compositional_field_types.size() == 1) ||
+                   (x_compositional_field_types.size() == n_compositional_fields),
+                   ExcMessage ("The length of the list of names for the field types of compositional "
+                               "fields needs to have one entry or have a length equal to "
+                               "the number of compositional fields."));
+
+      // If only one method is specified apply this to all fields
+      if (x_compositional_field_types.size() == 1)
+        x_compositional_field_types = std::vector<std::string> (n_compositional_fields, x_compositional_field_types[0]);
+
+      // If field types have been defined, fill the corresponding index vectors
+      if (x_compositional_field_types.size() == n_compositional_fields)
```

Comment:
Isn't this always true, because you extend it above?

Reply:
Agreed, and also at the moment you do not specify what happens for x_compositional_field_types.size() == 0. Wouldnt we want to do the same thing as when x_compositional_field_types.size() == 1, except setting it to `generic` for all entries?